compiler/src/dotty/tools/dotc/core/Names.scala

```diff
@@ -289,7 +290,32 @@ object Names {
 
     override def toString =
       if (length == 0) ""
-      else new String(chrs, start, length)
+      else {
+        if (Config.checkBackendNames) {
+          if (!toStringOK) {
+            println("Backend should not call Name#toString, Name#mangledString should be used instead.")
+            new Error().printStackTrace()
+            assert(false)
+          }
+        }
+        new String(chrs, start, length)
+      }
+
+    private def toStringOK = {
+      val trace = Thread.currentThread.getStackTrace
+      !trace.exists(_.getClassName.endsWith("GenBCode")) ||
+      trace.exists(elem =>
+          List(
+              "mangledString",
```

Comment:
I think that would affect too many things. Note that most of the whitelist does not call toString directly, sometimes there are 10-20 calls in between.